### DISCUSSION TITLE
Is macOS 15 missing from the repository readme?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [actions/virtual-environments#10472] [General questions] [answered]
Quote from:

- https://github.com/github/roadmap/issues/986

> To use macOS 15 directly, update the `runs-on:` key in you workflow file to `macos-15`, `macos-15-large`, or `macos-15-xlarge`

But the [readme on this repository](https://github.com/actions/runner-images/blob/main/README.md) does _not_ mention macOS 15 yet. Should the readme be updated, so it mentions the new macOS 15 runners?

**Comment by erik-bershel** [comment, marked as answer]
Hey @HonkingGoose!

Mentioned work item belongs to Q4 or Oct-Dec. macOS-15 is not published yet.

**Reply by HonkingGoose** [reply]
> macOS-15 is not published yet.

Whoops! You're right, it's not published yet!

Thank you for the quick response! ❤️

**Comment by ryanjwessel** [comment]
Following up on this thread, what is the expected SLA for adding support for the newest versions of MacOS when they are released? 

[The official release date is 9/16](https://9to5mac.com/2024/09/09/macos-sequoia-release-date/). 

My team would like to use some features, like support for dark mode and tinted app icons, that are only available in Xcode 16. But we'd prefer not to build from beta versions of software.

Is the suggested approach for teams to use the Xcode beta or the latest version from the previous MacOS version once Xcode 16 is generally available?

**Reply by erik-bershel** [reply]
New functionality is always available only in new versions of Xcode, Apple's release cycle implies that the new Xcode will be in beta for some time. Strictly speaking, no one forces anyone, but in fact, if you want your application to deliver new functionality before competitors, then you will have to use beta.

Xcode release history: https://xcodereleases.com/

**Reply by ryanjwessel** [reply]
Thanks, @erik-bershel. That makes sense and aligns with what we've had to do. At least Xcode 16 is compatible with MacOS Sonoma, so in theory, we could use the GA version once that is published.

Do you know what GitHub's SLA is for publishing new MacOS images? I'm just curious if the target is broadly Q4 or if the team aims to release a new image within 30/60 days of a new GA OS version release.

**Reply by erik-bershel** [reply]
Public target for macOS-15 is the end of Q3. Might be postponed for some time, but in this case we'll inform community via GitHub blog and/or roadmap.

**Reply by ryanjwessel** [reply]
Okay, thank you for that information, @erik-bershel; that is quite helpful for our planning!

**Comment by plajjan** [comment]
What sort of weird language tense is used in https://github.com/github/roadmap/issues/986 though!? It is written as if macos-15 is already done. That's literally what it says, but seeing how it is a roadmap item that's not completed, and the comment in this discussion, it seems like it's actually a future item. I suspect this is the summary text to be used once this is released...

The language used is really confusing. Could you please not write roadmap items like that? You are confusing the world. I googled "github actions macos 15" and found this, which led me to believe that macos-15 was indeed available already.

Instead of

> Actions: macOS 15 is now available for GitHub-hosted runners (GA)

Just avoid saying it's available, because, you know, it's not. Maybe like this?

> Actions: macos 15 for GitHub-hosted runners

Instead of
> macOS 15 is now generally available for GitHub-hosted runners. It will also become the latest macOS runner image with workflows migrating from November 2024 through December 2024.

Just avoid saying it's generally available when it's in fact not. Words have meaning.

**Comment by erik-bershel** [comment]
Psss, guys! 👋 
macOS-15 beta available here from today 🎉 

- https://github.com/actions/runner-images/issues/10686

cc: @HonkingGoose @ryanjwessel @plajjan
